src/generator.cs

```diff
@@ -4353,6 +4356,7 @@ void GenerateProperty (Type type, PropertyInfo pi, List<string> instance_fields_
 				PrintPlatformAttributes (pi);
 				PrintPlatformAttributes (pi.GetGetMethod ());
 				PrintPreserveAttribute (pi.GetGetMethod ());
+				PrintAdviceAttribute (pi.GetGetMethod ());
```

Comment:
maybe

> PrintAttributes (pi.GetGetMethod (), platform: true, preserve: true, advice:true); 

or we assume everything and only turn off out-of-context generation ?